src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueueFlushCondition.cs

```diff
@@ -7,13 +7,31 @@ namespace Stratis.Bitcoin.Features.BlockStore
     public sealed class BlockStoreQueueFlushCondition : IBlockStoreQueueFlushCondition
     {
         private readonly IChainState chainState;
+        private readonly IInitialBlockDownloadState blockDownloadState;
 
-        public BlockStoreQueueFlushCondition(IChainState chainState)
+        public BlockStoreQueueFlushCondition(IChainState chainState, IInitialBlockDownloadState blockDownloadState)
         {
             this.chainState = chainState;
+            this.blockDownloadState = blockDownloadState;
         }
 
         /// <inheritdoc/>
-        public bool ShouldFlush { get { return this.chainState.IsAtBestChainTip; } }
+        public bool ShouldFlush
+        {
+            get
+            {
+                // If the node is in IBD we don't flush on each block.
+                if (this.blockDownloadState.IsInitialBlockDownload())
+                    return false;
+
+                int distanceFromConsensusTip = this.chainState.ConsensusTip.Height - this.chainState.BlockStoreTip.Height;
+
+                // Once store is 5 blocks form the consensus tip then flush on every block.
+                if (distanceFromConsensusTip < 5)
```

Comment:
I'd still use `IsAtBestChainTip` over this

Reply:
But then we get the effect of flushing many blocks once tip is synced and tore is far behind.